internal/server/handle_test.go

```diff
@@ -0,0 +1,109 @@
+package server_test
+
+import (
+	"fmt"
+	"net/http/httptest"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/drykit-go/testx"
+	"github.com/drykit-go/testx/check"
+
+	"github.com/benchttp/cobaye/internal/server"
+)
+
+func TestHandleRequest(t *testing.T) {
+	t.Cleanup(server.ResetTotalRequest)
+
+	s := server.Server{}
+
+	t.Run("request with delay param", func(t *testing.T) {
+		const delay = 100
+		const expmin, expmax = delay * time.Millisecond, (delay + 10) * time.Millisecond
```

Comment:
This one fails on my cpu _sometimes_ (when it happens, it is only by 0.01s: at 0.11s).

Reply:
In short: only the handler's execution time

Technically, no request is sent: handlers are regular functions and we can unit test them by calling them with specific arguments (one of them being a request object) and see the results.
`drykit-go/testx` is an amazingly well though lib, as a consequence is only measures the execution time of the handler (just like you would for any regular function):
https://github.com/drykit-go/testx/blob/2f52657f1d91930702d638506c3c307f2f9acc36/runner_httphandler.go#L84-L86

Reply:
Ok thanks!
What an amazingly well thought lib, indeed 😄 
The PR is ok for me as it is then.